toolboxes/core/gpu/cuNDArray_blas.cu

```diff
@@ -1,312 +1,504 @@
+    }
     return result;
-  }
-  
-  template<class T> size_t amin( cuNDArray<T>* x )
-  {
-    if( x == 0x0 )
-      throw std::runtime_error("Gadgetron::amin(): Invalid input array");
+}
+
+template <typename T> size_t amin(cuNDArray<T>* x) {
+    if (x == 0x0)
+        throw std::runtime_error("Gadgetron::amin(): Invalid input array");
+
+    int device = cudaDeviceManager::Instance()->getCurrentDevice();
+    size_t result;
+
+    // If number of elements in the array is greater than INT_MAX break it up and perform calculations this is done to
+    // support large data arrays
+
+    int num_splits = x->get_number_of_elements() / INT_MAX + 1;
+    int remainder = x->get_number_of_elements() - INT_MAX * (num_splits - 1);
+
+    for (int ii = 0; ii < num_splits; ii++) {
+
+        int interim_result;
+
+        CUBLAS_CALL(cublas_amin(cudaDeviceManager::Instance()->lockHandle(device),
+                                (ii == num_splits - 1) ? remainder : INT_MAX, // n number of elements
+                                x->get_data_ptr() + INT_MAX * ii, 1, &interim_result));
+
+        cudaDeviceManager::Instance()->unlockHandle(device);
+
+        if (ii == 0)
+            result = (size_t)interim_result - 1;
+        else if (((*x)[INT_MAX * ii + (size_t)interim_result - 1]) < ((*x)[result]))
+            result = INT_MAX * ii + (size_t)interim_result - 1;
+
+        if (result > x->get_number_of_elements()) {
+            throw std::runtime_error("Gadgetron::amin(): computed index is out of bounds");
+        }
+    }
+    return result; // result - 1;
+}
+
+template <typename T> size_t amax(cuNDArray<T>* x) {
+    if (x == 0x0)
+        throw std::runtime_error("Gadgetron::amax(): Invalid input array");
 
     int device = cudaDeviceManager::Instance()->getCurrentDevice();
-    int result;
+    size_t result;
+
+    // If number of elements in the array is greater than INT_MAX break it up and perform calculations this is done to
+    // support large data arrays
+
+    int num_splits = x->get_number_of_elements() / INT_MAX + 1;
+    int remainder = x->get_number_of_elements() - INT_MAX * (num_splits - 1);
+
+    for (int ii = 0; ii < num_splits; ii++) {
+
+        int interim_result;
+        CUBLAS_CALL(cublas_amax(
+            cudaDeviceManager::Instance()->lockHandle(device),
+            (ii == num_splits - 1) ? remainder : INT_MAX, // n number of elements (int)x->get_number_of_elements(),
+            x->get_data_ptr() + INT_MAX * ii, 1, &interim_result));
 
-    CUBLAS_CALL(cublas_amin(cudaDeviceManager::Instance()->lockHandle(device),
-                            (int)x->get_number_of_elements(), x->get_data_ptr(), 1, &result));
+        cudaDeviceManager::Instance()->unlockHandle(device);
+       
+        if (ii == 0)
+            result = (size_t)interim_result - 1;
+        else if ((((*x)[INT_MAX * ii + (size_t)interim_result - 1]) > ((*x)[result])))
+            result = INT_MAX * ii + (size_t)interim_result - 1;
```

Comment:
What's the desired result in the case where ii != 0 and (((*x)[INT_MAX * ii + (size_t)interim_result - 1]) <((*x)[result])))?

Reply:
It should fail.